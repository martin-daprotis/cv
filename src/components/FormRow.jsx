import React from 'react'
import BorderLinearProgress from "./BorderLinearProgress";
import ChipColor from "./ChipColor";
import { makeStyles } from "@material-ui/core/styles";
import { Grid,  Typography } from "@material-ui/core";
import "react-circular-progressbar/dist/styles.css";

const useStyles = makeStyles((theme) => ({
  item: {
    color: theme.palette.text.primary,
  },
}));

const FormRow = ({ t,items, startAnimation }) => {
const classes = useStyles();

  return (
    <>
      {items.map((it, i) => {
        return (
          <Grid
            container
            key={it.title}
            item
            style={{ marginBottom: "10px", paddingLeft: "15px", paddingRight: "15px" }}
          >
            <Grid item xs={6} className={classes.item}>
              <ChipColor
                title={t(it.title)}
                bgColor={"#dce775"}
                icon={it.icon}
              />
            </Grid>
            <Grid item xs={6}>
              <Typography align="right">{it.value}%</Typography>
            </Grid>
            <Grid item xs={12} style={{ marginTop: "5px" }}>
              <BorderLinearProgress startAnimation={startAnimation} value={it.value} bgColor={"#dce775"} />
            </Grid>
          </Grid>
        );
      })}
    </>
  );
}

export default FormRow;