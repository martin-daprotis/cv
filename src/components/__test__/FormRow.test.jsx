import React from 'react'
import { render, fireEvent, cleanup,logDOM} from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import {
  translate,
} from "react-switch-lang";
import FormRow from '../FormRow';


const mockSkills = [
      { "title": "JavaScript", "icon": "devicon-javascript-plain", "value": 80 },
      { "title": "React", "icon": "cib-react", "value": 100 },
    ];

afterEach(cleanup)

it('render every item ', () => {
  const RenderHOC = translate(({t}) => <FormRow  items={mockSkills} t={t} startAnimation='false'/>);
  const {getByText} = render(<RenderHOC/>)
  expect(getByText('JavaScript')).toBeInTheDocument();
  expect(getByText('React')).toBeInTheDocument();
})

it('should render percentage ', () => {
  const RenderHOC = translate(({t}) => <FormRow  items={mockSkills} t={t} startAnimation='true'/>);
  const {getByText} = render(<RenderHOC/>)
  expect(getByText('100%')).toBeInTheDocument();
})

it('matches snapshot',  () => {
  const RenderHOC = translate(({t}) =><FormRow  items={mockSkills} t={t} startAnimation='true'/>);
  const wrapper = render(<RenderHOC/>)
  expect(wrapper.asFragment()).toMatchSnapshot()
})