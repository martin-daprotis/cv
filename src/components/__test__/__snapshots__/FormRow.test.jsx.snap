// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item"
    style="margin-bottom: 10px; padding-left: 15px; padding-right: 15px;"
  >
    <div
      class="MuiGrid-root makeStyles-item-27 MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar makeStyles-avatar-28 makeStyles-avatar-30 MuiChip-avatarSmall MuiAvatar-colorDefault"
        >
          <span
            class="devicon-javascript-plain"
            style="font-size: 1.3em; color: rgb(255, 255, 255);"
          />
        </div>
        <span
          class="MuiChip-label MuiChip-labelSmall"
        >
          JavaScript
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
      >
        80%
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      style="margin-top: 5px;"
    >
      <div
        class=" makeStyles-root-31"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="MuiLinearProgress-root makeStyles-root-33 makeStyles-root-35 MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar makeStyles-bar-34 makeStyles-bar-36 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item"
    style="margin-bottom: 10px; padding-left: 15px; padding-right: 15px;"
  >
    <div
      class="MuiGrid-root makeStyles-item-27 MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar makeStyles-avatar-28 makeStyles-avatar-37 MuiChip-avatarSmall MuiAvatar-colorDefault"
        >
          <span
            class="cib-react"
            style="font-size: 1.3em; color: rgb(255, 255, 255);"
          />
        </div>
        <span
          class="MuiChip-label MuiChip-labelSmall"
        >
          React
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
      >
        100%
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      style="margin-top: 5px;"
    >
      <div
        class=" makeStyles-root-31"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="MuiLinearProgress-root makeStyles-root-33 makeStyles-root-38 MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar makeStyles-bar-34 makeStyles-bar-39 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
